### PR TITLE
Issue #211 [Title] Support links for titles.

### LINKED ISSUE #211: [Title] Support links for titles
It would be useful if the title would have support for links. The link would be optional and would allow titles to link either to an internal page in AEM, an external URL or an anchor.

### PR DESCRIPTION
**Pull request by Shivanshmathur**
Added Hobbes test for testing link
Removed spaces in Title.html

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #211  
| Patch: Bug Fix?          | Yes
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes- Added Hobbes Test
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

The html renderer couldn't handle the spaces in html so removed the spaces from the tags containing text.

### CODE REVIEW

**Comment by Shivanshmathur** [comment]
@richardhand added the Hobbes test for links.

File: testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/Title/v1/Title.js [hunk 1]
```diff
@@ -267,4 +267,33 @@ window.CQ.CoreComponentsIT.Title.v1 = window.CQ.CoreComponentsIT.Title.v1 || {}
...
+     */
+    title.tcSetLink = function(tcExecuteBeforeTest, tcExecuteAfterTest) {
+        return new h.TestCase("Set Link", {
+            execBefore: tcExecuteBeforeTest,
+            execAfter: tcExecuteAfterTest })
+
+            // open the config dialog
+            .execTestCase(c.tcOpenConfigureDialog("cmpPath"))
+            // enter the link
+            .simulate("foundation-autocomplete[name='./linkURL'] input[type!='hidden']", "key-sequence",
```
Comment (richardhand) [inline testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/Title/v1/Title.js]: would rather use `foundation-autocomplete[name='./linkURL'] input[is="coral-textfield"]` as selector here.

File: testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/Title/v1/Title.js [hunk 2]
```diff
@@ -267,4 +267,33 @@ window.CQ.CoreComponentsIT.Title.v1 = window.CQ.CoreComponentsIT.Title.v1 || {}
...
+                { sequence: c.rootPage + "{enter}" })
+            // save the dialog
+            .execTestCase(c.tcSaveConfigureDialog)
+
+            // switch to content frame
+            .config.changeContext(c.getContentFrame)
+            // click on the title
+            .click("div.cmp-title a", { expectNav: true })
```
Comment (richardhand) [inline testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/Title/v1/Title.js]: and here, let's use the BEM element, `.cmp-title__link`

File: testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/Title/v1/Title.js (hunk 1 above)
Comment (Shivanshmathur) [reply testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/Title/v1/Title.js]: Done.

File: testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/Title/v1/Title.js (hunk 2 above)
Comment (Shivanshmathur) [reply testing/it/ui-js/src/content/jcr_root/apps/core/wcm/tests/test-cases/Title/v1/Title.js]: Done

**Pull request merged**